backend/src/test/java/org/west/backend/controller/GameControllerTest.java

```diff
@@ -59,4 +60,29 @@ void deleteGameById() throws Exception {
 
         verify(mockGameRepository).deleteById("1");
     }
+
+    @Test
+    void addGame() throws Exception {
+        when(mockGameRepository.save(gameTestData.getFirst())).thenReturn(gameTestData.getFirst());
```

Comment:
Wozu wird der When benötigt?
Du gibst die Daten schon mit, die als Result abgetestet werden:
mockMvc.perform(MockMvcRequestBuilders.post("/api/wishlist")
                .contentType(MediaType.APPLICATION_JSON)
                .content("""
                            {
                               "id": "1",
                               "name": "The Legend of Zelda: Breath of the Wild",
                               "releaseDate": "2017-03-03",
                               "fav": true
                            }
                        """))
Ich glaube, dass when wird nicht benötigt.

Reply:
Ich hab es ohne das when probiert, allerdings ist dann der Body-Response leer. Wahrscheinlich, weil nichts in der Datenbank enthalten ist.

Das ist die Erklärung von ChatGPT:

In Ihrem konkreten Testfall:

    Sie definieren, dass das Speichern eines bestimmten Spiels (gameTestData.getFirst()) im mockGameRepository das gleiche Spielobjekt zurückgeben soll.
    Dies stellt sicher, dass der Controller-Test (mockMvc.perform) das erwartete Verhalten auslöst, wenn der POST-Aufruf auf /api/wishlist gemacht wird.

Zusammengefasst wird der when-Block verwendet, um das Mock-Objekt zu konfigurieren und das erwartete Verhalten der Abhängigkeiten in Ihrem Unit-Test festzulegen.

Aber ich kann deinen Einwand gut verstehen, ich hab mich nämlich dumm und dämlich probiert ohne das when ^^"